mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -1585,13 +1585,25 @@ OpFoldResult ReshapeOp::fold(FoldAdaptor adaptor) {
           getResult().getType()))
     return reshapedSource;
 
+  // If the producer of operand 'source' is another 'tensor.reshape' op, use the
+  // producer's input instead as the original tensor to reshape. This could
+  // render such producer dead code.
+  if (auto reshapeOpProducer = getSource().getDefiningOp<ReshapeOp>()) {
+    setOperand(0, reshapeOpProducer.getSource());
```

Comment:
This might be clearer/safer using the named accessor, rather than `setOperand`:

```cpp
getSourceMutable().assign(reshapeOpProducer.getSource());
```